northstar/src/runtime/state.rs

```diff
@@ -166,24 +193,44 @@ impl<'a> State<'a> {
                 .add_buf(hello_world)
                 .await
                 .expect("Failed to load hello-world");
-            repositories.insert(INTERNAL_REPOSITORY.into(), Box::new(internal));
+            self.repositories
+                .insert(INTERNAL_REPOSITORY.into(), Box::new(internal));
         }
 
-        // TODO: Verify that the containers in all repositories are unique with name and version
+        Ok(())
+    }
 
-        let launcher_island = Island::start(events_tx.clone(), config.clone())
-            .await
-            .expect("Failed to start launcher");
-        let mount_control = MountControl::new(&config).await.map_err(Error::Mount)?;
+    async fn autostart(&mut self) -> Result<(), Error> {
+        // List of containers from all repositories with the autostart flag set
+        let autostart = self
+            .repositories
+            .iter()
+            .map(|(_, r)| r.containers())
+            .flatten()
+            .filter(|n| n.manifest().autostart)
+            .map(|n| Container::new(n.manifest().name.clone(), n.manifest().version.clone()))
+            .collect::<Vec<_>>();
 
-        Ok(State {
-            events_tx,
-            repositories,
-            containers: HashMap::new(),
-            config,
-            launcher_island,
-            mount_control: Arc::new(mount_control),
-        })
+        // Mount (parallel)
+        let mounts = self.mount_all(&autostart).await;
+
+        for (result, container) in mounts.iter().zip(autostart) {
+            match result {
+                Ok(_) => {
+                    info!("Autostarting {}", container);
+                    if let Err(e) = self.start(&container).await {
+                        warn!("Failed to autostart {}: {}", container, e);
+                        // TODO: What shall we do with this error?
```

Comment:
I think autostart is done on a best-effort basis so printing a warning and continuing is sensible.

Reply:
I don't agree. Containers flagged with `autostart` are something that is *not* started by someone and therefore probably *nobody* takes care about them. Image a "root" container that takes care for starting all other containers. If this root container doesn't start, the whole system is probably broken. 

What about extending the `autostart` flags:
```rust
enum Autostart {
  /// Ignore errors when starting this application. Ignore the containers termination
  Relaxed,
  /// If the container doesn't start or exits with a non zero exit code the runtime exits with a failure
  Critical,
  /// Treat a failure when starting this container as a critical incident and terminate the runtime. Ignore any termination o the container
  CriticalStart,
}
...
autostart: Autostart,
```

Reply:
@marcmo @inorick any opinion?